kyo-core/shared/src/main/scala/kyo/Fiber.scala

```diff
@@ -396,7 +394,7 @@ object Fiber extends FiberPlatformSpecific:
                                     quickSort(indices, results, size)
 
                                     // Limit final result to max successful results
-                                    completeDiscard(Result.succeed(Chunk.fromNoCopy(results).take(size)))
+                                    completeDiscard(Result.succeed(Chunk.fromNoCopy(results).take(size).asInstanceOf[Chunk[A]]))
```

Comment:
The underlying `results` is untyped so this produces `Chunk[AnyRef]` but it's ok to cast here